runtime/queries/table_columns.go

```diff
@@ -47,61 +47,69 @@ func (q *TableColumns) Resolve(ctx context.Context, rt *runtime.Runtime, instanc
 		return fmt.Errorf("not available for dialect '%s'", olap.Dialect())
 	}
 
-	temporaryTableName := tempName("profile_columns_")
-	// views return duplicate column names, so we need to create a temporary table
-	rows, err := olap.Execute(ctx, &drivers.Statement{
-		Query:    fmt.Sprintf(`CREATE TEMPORARY TABLE "%s" AS (SELECT * FROM "%s" LIMIT 1)`, temporaryTableName, q.TableName),
-		Priority: priority,
-	})
-	if err != nil {
-		return err
-	}
-	rows.Close()
-	defer dropTempTable(olap, priority, temporaryTableName)
+	return olap.WithConnection(ctx, priority, func(ctx context.Context, ensuredCtx context.Context) error {
+		// views return duplicate column names, so we need to create a temporary table
+		temporaryTableName := tempName("profile_columns_")
+		err = olap.Exec(ctx, &drivers.Statement{
+			Query:    fmt.Sprintf(`CREATE TEMPORARY TABLE "%s" AS (SELECT * FROM "%s" LIMIT 1)`, temporaryTableName, q.TableName),
+			Priority: priority,
+		})
+		if err != nil {
+			return err
+		}
+		defer func() {
+			// NOTE: Using ensuredCtx
+			_ = olap.Exec(ensuredCtx, &drivers.Statement{
+				Query:    `DROP TABLE "` + temporaryTableName + `"`,
+				Priority: priority,
+			})
+		}()
 
-	rows, err = olap.Execute(ctx, &drivers.Statement{
-		Query: fmt.Sprintf(`select column_name as name, data_type as type from information_schema.columns 
+		rows, err := olap.Execute(ctx, &drivers.Statement{
+			Query: fmt.Sprintf(`select column_name as name, data_type as type from information_schema.columns 
 		where table_name = '%s' and table_schema = 'temp'`, temporaryTableName),
-		Priority: priority,
-	})
-	if err != nil {
-		return err
-	}
-	defer rows.Close()
-	var pcs []*runtimev1.ProfileColumn
-	i := 0
-	for rows.Next() {
-		pc := runtimev1.ProfileColumn{}
-		if err := rows.StructScan(&pc); err != nil {
+			Priority: priority,
+		})
+		if err != nil {
 			return err
 		}
-		pcs = append(pcs, &pc)
-		i++
-	}
+		defer rows.Close()
```

Comment:
should we catch and log error from `rows.Close()` instead of ignoring

Reply:
In a perfect world, yeah. We have this many places, we can add a backlog item for it